sensors/sensors.go

```diff
@@ -125,8 +217,41 @@ func (lidar Lidar) TimedSensorReading(ctx context.Context) (TimedSensorReadingRe
 		readingTime, err = time.Parse(time.RFC3339Nano, timeRequestedMetadata[0])
 		if err != nil {
 			msg := "replay sensor timestamp parse RFC3339Nano error"
-			return TimedSensorReadingResponse{}, errors.Wrap(err, msg)
+			return TimedLidarSensorReadingResponse{}, errors.Wrap(err, msg)
 		}
 	}
-	return TimedSensorReadingResponse{Reading: buf.Bytes(), ReadingTime: readingTime, Replay: replay}, nil
+	return TimedLidarSensorReadingResponse{Reading: buf.Bytes(), ReadingTime: readingTime, Replay: replay}, nil
+}
+
+// TimedIMUSensorReading returns data from the IMU movement sensor and the time the reading is from.
+// IMU Sensors currently do not support replay capabilities.
+func (imu IMU) TimedIMUSensorReading(ctx context.Context) (TimedIMUSensorReadingResponse, error) {
+	replay := false
+	ctxWithMetadata, md := contextutils.ContextWithMetadata(ctx)
+	linAcc, err := imu.imu.LinearAcceleration(ctxWithMetadata, make(map[string]interface{}))
+	if err != nil {
+		msg := "LinearAcceleration error"
+		return TimedIMUSensorReadingResponse{}, errors.Wrap(err, msg)
+	}
+	angVel, err := imu.imu.AngularVelocity(ctxWithMetadata, make(map[string]interface{}))
+	if err != nil {
+		msg := "AngularVelocity error"
+		return TimedIMUSensorReadingResponse{}, errors.Wrap(err, msg)
+	}
+	readingTime := time.Now().UTC()
+
+	timeRequestedMetadata, ok := md[contextutils.TimeRequestedMetadataKey]
```

Comment:
[q/note] This is to check if the sensor is a replay camera, correct?

Reply:
Yes